specification/app/resource-manager/Microsoft.App/preview/2023-08-01-preview/Patches.json

```diff
@@ -0,0 +1,542 @@
+            "schema": {
+              "$ref": "#/definitions/PatchResource"
+            }
+          },
+          "202": {
+            "description": "Patching operation has been started.",
+            "schema": {
+              "$ref": "#/definitions/PatchResource"
+            },
+            "headers": {
+              "Azure-AsyncOperation": {
+                "type": "string"
+              }
+            }
+          },
+          "default": {
+            "description": "Common error response.",
+            "schema": {
+              "$ref": "./CommonDefinitions.json#/definitions/DefaultErrorResponse"
+            }
+          }
+        },
+        "x-ms-long-running-operation-options": {
+          "final-state-via": "azure-async-operation"
+        },
+        "x-ms-long-running-operation": true,
+        "x-ms-examples": {
+          "Patches_Apply_0": {
+            "$ref": "./examples/Patches_Apply.json"
+          }
+        }
+      }
+    }
+  },
+  "definitions": {
+    "PatchResource": {
+      "type": "object",
+      "properties": {
+        "properties": {
+          "$ref": "#/definitions/PatchProperties",
+          "description": "The resource-specific properties for this resource.",
+          "x-ms-client-flatten": true,
+          "x-ms-mutability": [
+            "read",
+            "create"
+          ]
+        }
+      },
+      "description": "The resource representation of a patch.",
+      "allOf": [
+        {
+          "$ref": "../../../../../common-types/resource-management/v3/types.json#/definitions/ProxyResource"
+        }
+      ]
+    },
+    "PatchProvisioningState": {
+      "type": "string",
+      "description": "Resource instance provisioning state.",
+      "enum": [
+        "Succeeded",
+        "Failed",
+        "Canceled"
+      ],
+      "x-ms-enum": {
+        "name": "PatchProvisioningState",
+        "modelAsString": true
+      },
+      "readOnly": true
+    },
+    "PatchApplyStatus": {
+      "type": "string",
+      "description": "The status of the patch once it has been provisioned",
+      "enum": [
+        "NotStarted",
+        "RebaseInProgress",
+        "CreatingRevision",
+        "Succeeded",
+        "Canceled",
+        "RebaseFailed",
+        "RevisionCreationFailed",
+        "ImagePushPullFailed",
+        "ManuallySkipped"
+      ],
+      "x-nullable": true,
+      "x-ms-enum": {
+        "name": "PatchApplyStatus",
+        "modelAsString": true
+      },
+      "readOnly": true
+    },
+    "DetectionStatus": {
+      "type": "string",
+      "description": "The status of the patch detection.",
+      "enum": [
+        "Succeeded",
+        "RegistryLoginFailed",
+        "Failed"
+      ],
+      "x-ms-enum": {
+        "name": "DetectionStatus",
+        "modelAsString": true
+      },
+      "readOnly": true
+    },
+    "PatchDetails": {
+      "description": "The details of specific patch item.",
```

Comment:
>     "description": "The details of specific patch item.", [](http://example.com/codeflow?start=1&length=58)

what one? How does this compare with PatchProperties?

Reply:
sorry for the confusion. The patch details are the details of the patching operations going to be performed when user applies the patch. There could be multiple patch details if there are multiple containers in a container app revision. The patch properties describes the entire patch resource as a whole. 

*we have a 1:1 ratio of the patch resource and the container app revision, and 1:1 ratio of the number of patch detail and the patchable containers in one revision.